addons/web/static/src/notifications/notification_service.js

```diff
@@ -13,11 +13,13 @@ export const notificationService = {
     let notifications = [];
     const bus = new EventBus();
 
-    function close(id) {
+    function close(id, wait = 0) {
       const index = notifications.findIndex((n) => n.id === id);
       if (index > -1) {
-        notifications.splice(index, 1);
-        bus.trigger("UPDATE", notifications);
+        browser.setTimeout(() => {
```

Comment:
I would do everything inside the setTimeout, because if the notifications list changes meanwhile, you might do something unexpected